examples/pwm.rs

```diff
@@ -24,7 +24,7 @@ fn main() -> ! {
     let gpioa = dp.GPIOA.split(&mut rcc);
 
     // Initialize TIM2 for PWM
-    let pwm = pwm::Timer::new(dp.TIM2, 10.khz(), &mut rcc);
+    let pwm = pwm::Timer::new(dp.TIM2, 10_000.Hz(), &mut rcc);
```

Comment:
This (and IIRC the SPI interface require this change unless I use `TryInto` in the APIs - embedded-time can convert from e.g. `Khz<u32>` to `Hz<u64>` without failing, but needs the fallible conversions if we want to go from `Khz<u32>` to `Hz<u32>`. Realistically, confining to `u32` everywhere isn't an issue (IMO) so I'd be up for using `TryInto` and clamping to `u32::MAX` like I do with the `delay()` API. Thoughts?